core/primitives/piece/impl/piece_data.cpp

```diff
@@ -8,11 +8,13 @@
 #include <boost/filesystem.hpp>
 
 namespace fc::primitives::piece {
-  PieceData::PieceData(const std::string &path_to_file, int flags) {
+  PieceData::PieceData(const std::string &path_to_file, int flags)
+      : isNullData(false) {
     fd_ = open(path_to_file.c_str(), flags, 0644);
   }
 
   int PieceData::getFd() const {
+    assert(not isNullData);
```

Comment:
@Markuu-s make sure it doesn't affect all places where `getFd()` is called, even tests

Reply:
It affects. I'll see what I can do with it.

Reply:
Fixed